hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientReplicatedMapProxy.java

```
@@ -112,7 +112,11 @@ protected void onInitialize() {
     private void initNearCache() {
         NearCacheConfig nearCacheConfig = getContext().getClientConfig().getNearCacheConfig(name);
         if (nearCacheConfig != null) {
-            nearCache = getContext().getNearCacheManager(getServiceName()).getOrCreateNearCache(name, nearCacheConfig);
+            NearCacheConfig copyNearCacheConfig = new NearCacheConfig(nearCacheConfig);
+            //We don't have serializeKeys support on replicated map nearcache
+            copyNearCacheConfig.setSerializeKeys(false);
```

can we fail fast here instead of forcing it to be `false`?